src/NuGet.Core/NuGet.Commands/SourcesCommands/SourcesRunner.cs

```diff
@@ -0,0 +1,299 @@
+                case SourcesAction.List:
+                    switch (args.Format)
+                    {
+                        case SourcesListFormat.Short:
+                            PrintRegisteredSourcesShort(args);
+                            break;
+                        case SourcesListFormat.Detailed:
+                        default:
+                            PrintRegisteredSourcesDetailed(args);
+                            break;
+                    }
+                    break;
+                case SourcesAction.Add:
+                    AddNewSource(args);
+                    break;
+                case SourcesAction.Remove:
+                    RemoveSource(args);
+                    break;
+                case SourcesAction.Enable:
+                    EnableOrDisableSource(args, enabled: true);
+                    break;
+                case SourcesAction.Disable:
+                    EnableOrDisableSource(args, enabled: false);
+                    break;
+                case SourcesAction.Update:
+                    UpdatePackageSource(args);
+                    break;
+            }
+        }
+
+        private static void EnableOrDisableSource(SourcesArgs args, bool enabled)
+        {
+            if (string.IsNullOrEmpty(args.Name))
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandNameRequired));
+            }
+
+            var packageSource = args.SourceProvider.GetPackageSourceByName(args.Name);
+            if (packageSource == null)
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandNoMatchingSourcesFound, args.Name));
+            }
+
+            if (enabled && !packageSource.IsEnabled)
+            {
+                args.SourceProvider.EnablePackageSource(args.Name);
+            }
+            else if (!enabled && packageSource.IsEnabled)
+            {
+                args.SourceProvider.DisablePackageSource(args.Name);
+            }
+
+            if (enabled)
+            {
+                args.LogMinimal(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandSourceEnabledSuccessfully, args.Name));
+            }
+            else
+            {
+                args.LogMinimal(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandSourceDisabledSuccessfully, args.Name));
+            }
+        }
+
+        private static void RemoveSource(SourcesArgs args)
+        {
+            if (string.IsNullOrEmpty(args.Name))
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandNameRequired));
+            }
+
+            // Check to see if we already have a registered source with the same name or source
+            var sourceList = args.SourceProvider.LoadPackageSources().ToList();
```

Comment:
Redundant line of code as there are no references to `sourceList` in this method.

Reply:
boilerplate reply: the code in SourcesRunner.cs was moved from SourcesCommand.cs - I have tried to make the minimal changes to it. That said, I'll review each of these comments. thx.

Reply:
fixed in next commit.